### PR TITLE
FOGL-6619 Catch errors in Python and improve code structure

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

**Review by ashwinscale — CHANGES_REQUESTED** [review]
Review by test. Unable to push good data.

File: python35_filter.cpp [hunk 1]
```diff
@@ -25,6 +27,224 @@
...
+	m_init = true;
+
+	PyGILState_STATE state = PyGILState_Ensure(); // acquire GIL
+
+	// Pass Fledge Data dir
+	setFiltersPath(getDataDir());
+
+	// Set Python path for embedded Python 3.5
+	// Get current sys.path. borrowed reference
+	PyObject* sysPath = PySys_GetObject((char *)string("path").c_str());
```
Comment (AmandeepArora) [inline python35_filter.cpp:52]: Do we need to create a string object, can't we just use ' (char *)"path" ' ?

File: python35_filter.cpp [hunk 2]
```diff
@@ -136,50 +356,76 @@ vector<Reading *>* Python35Filter::getFilteredReadings(PyObject* filteredData)
...
+		{
+			m_logger->error("Python error: %s in supplied script", err_msg);
+		}
+		else
+		{
+			m_logger->error("Python error: %s in %s at line %s of supplied script", err_msg, error_line, actual_line_no);
+		}
+
+		PyErr_Clear();
+	}
```
Comment (AmandeepArora) [inline python35_filter.cpp:428]: I think we would need the Py_CLEAR statements for the various python objects used here.

**Review by AmandeepArora — COMMENTED** [review]
Please check for any missing Py_CLEAR statements in other changed code blocks also.

**Review by ashwinscale — APPROVED** [review]
LGTM Review by test

File: python35_filter.cpp (hunk 1 above)
Comment (MarkRiddoch) [reply python35_filter.cpp:52]: I suspect we don't this is old code that can probably be removed but I would rather do that sort of thing in a future change where we are not in as much of a rush to satisfy a customer.

File: python35_filter.cpp (hunk 2 above)
Comment (MarkRiddoch) [reply python35_filter.cpp:428]: Well spotted

**Review by AmandeepArora — APPROVED** [review]

**Pull request merged**